activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -120,12 +125,16 @@ def initialize(*addresses)
           raise ArgumentError, "First argument must be an empty array, an array of hosts or a Dalli::Client instance."
```

Comment:
```suggestion
          raise ArgumentError, "First argument must be an address or array of addresses."
```

1. Let's have this PR remove the suggestion of using a Dalli::Client instance.

and since we're doing that...

2. "An empty array" (or nil) is not going to be useful so let's not suggest that either. (I'm not sure why that's not an error, but that's out of scope.)
3. The docs call them "addresses" and not "hosts" so let's use that term.

Reply:
Oh I see, `nil` ends up being the same as `[]` due to a `.compact`:

https://github.com/rails/rails/blob/e184d33dfcef9e7265781ec84af559af4781765c/activesupport/lib/active_support/cache/mem_cache_store.rb#L93

Which ends up using Dalli's defaults:

https://github.com/petergoldstein/dalli/blob/8ad238df762b5bccf1b19343f712c8d1e23656cb/lib/dalli/servers_arg_normalizer.rb#L43

So "an empty array" (or nil) is fine. That was my misunderstanding, sorry.